File: app/src/main/java/com/musicseque/artist/activity/ReportProblemActivity.java
package com.musicseque.artist.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.musicseque.MainActivity;
import com.musicseque.R;
import com.musicseque.activities.BaseActivity;
import com.musicseque.artist.fragments.ProfileDetailFragment;
import com.musicseque.interfaces.MyInterface;
import com.musicseque.retrofit_interface.ImageUploadClass;
import com.musicseque.start_up.LoginActivity;
import com.musicseque.utilities.Constants;
import com.musicseque.utilities.Utils;

import org.json.JSONException;
import org.json.JSONObject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import okhttp3.MediaType;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;

public class ReportProblemActivity extends BaseActivity implements MyInterface {
    @BindView(R.id.img_first_icon)
    ImageView img_first_icon;
    @BindView(R.id.img_right_icon)
    ImageView img_right_icon;
    @BindView(R.id.tv_title)
    TextView tv_title;

    @BindView(R.id.rlAttachment)
    RelativeLayout rlAttachment;

    @BindView(R.id.tvEmail)
    TextView tvEmail;

    @BindView(R.id.tvImageName)
    TextView tvImageName;

    @BindView(R.id.etMessage)
    EditText etMessage;

    String mEmail, mMessage;
    private MultipartBody.Part fileUpload;
    private RequestBody mID;
    private String mFileName ;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_report_problem);
        ButterKnife.bind(this);
        tv_title.setText("Report A Problem");
        img_first_icon.setVisibility(View.VISIBLE);
        img_right_icon.setVisibility(View.GONE);

        tvEmail.setText(sharedPreferences.getString(Constants.EMAIL_ID, ""));

    }


    @OnClick({R.id.img_first_icon, R.id.btnSubmit, R.id.txt_attach})
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.img_first_icon:
                finish();
                break;
            case R.id.btnSubmit:
                mEmail = tvEmail.getText().toString();
                mMessage = etMessage.getText().toString();
                if (mEmail.length() == 0) {
                    Utils.showToast(ReportProblemActivity.this, getResources().getString(R.string.err_email_id));
                } else if (!Utils.emailValidator(mEmail)) {
                    Utils.showToast(ReportProblemActivity.this, getResources().getString(R.string.err_email_invalid));
                } else if (mMessage.length() == 0) {
                    Utils.showToast(ReportProblemActivity.this, "Please enter the message");
                } else {
                    if (Utils.isNetworkConnected(ReportProblemActivity.this)) {
                        Utils.initializeAndShow(ReportProblemActivity.this);
                        hitAPI();
                    } else {
                        Utils.showToast(ReportProblemActivity.this, getResources().getString(R.string.err_no_internet));

                    }
                }

                break;
            case R.id.txt_attach:
                openDialog("report");
                break;
        }
    }

    private void hitAPI() {
        RequestBody mUSerId = RequestBody.create(MediaType.parse("text/plain"), sharedPreferences.getString(Constants.USER_ID, ""));
        RequestBody message = RequestBody.create(MediaType.parse("text/plain"), mMessage);
        ImageUploadClass.imageUpload(fileUpload, mUSerId, message, Constants.FOR_REPORT_PROBLEM, ReportProblemActivity.this);


    }

    public void getImageDetails(MultipartBody.Part fileToUpload, RequestBody mUSerId, String name) {
        fileUpload = fileToUpload;
        mFileName=name;
        tvImageName.setVisibility(View.VISIBLE);
        tvImageName.setText(name);
        // mID = mUSerId;
        // hitAPI();

    }

    @Override
    public void sendResponse(Object response, int TYPE) {
        Utils.hideProgressDialog();
        try {
            JSONObject jsonObject = new JSONObject(response.toString());
            if (jsonObject.getString("Status").equalsIgnoreCase("success")) {
                Utils.showToast(ReportProblemActivity.this, jsonObject.getString("Message"));
                Intent intent = new Intent(ReportProblemActivity.this, MainActivity.class);

                startActivity(intent);
                finish();
            } else {
                Utils.showToast(ReportProblemActivity.this, jsonObject.getString("Message"));
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/musicseque/artist/activity/ReportProblemActivity.kt
package com.musicseque.artist.activity

import android.content.Intent
import android.os.Bundle
import android.view.View

import com.musicseque.MainActivity
import com.musicseque.R
import com.musicseque.activities.BaseActivity
import com.musicseque.interfaces.MyInterface
import com.musicseque.retrofit_interface.ImageUploadClass
import com.musicseque.utilities.Constants
import com.musicseque.utilities.SharedPref
import com.musicseque.utilities.Utils
import kotlinx.android.synthetic.main.activity_report_problem.*
import kotlinx.android.synthetic.main.toolbar_top.*
import okhttp3.MediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody
import org.json.JSONException
import org.json.JSONObject

class ReportProblemActivity : BaseActivity(), MyInterface, View.OnClickListener {

    private var mMessage: String = ""
    private var mEmail: String = ""
    private var fileUpload: MultipartBody.Part? = null
    private val mID: RequestBody? = null
    private var mFileName: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_report_problem)
        initViews()
        listeners()

    }

    private fun initViews() {
        tv_title!!.text = "Report A Problem"
        img_first_icon!!.visibility = View.VISIBLE
        img_right_icon!!.visibility = View.GONE
        tvEmail!!.text = SharedPref.getString(Constants.EMAIL_ID, "")

    }

    private fun listeners() {
        img_first_icon.setOnClickListener(this)
        btnSubmit.setOnClickListener(this)
        txt_attach.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.img_first_icon -> finish()
            R.id.btnSubmit -> {
                mEmail = tvEmail!!.text.toString()
                mMessage = etMessage!!.text.toString()
                if (mEmail!!.length == 0) {
                    Utils.showToast(this@ReportProblemActivity, resources.getString(R.string.err_email_id))
                } else if (!Utils.emailValidator(mEmail)) {
                    Utils.showToast(this@ReportProblemActivity, resources.getString(R.string.err_email_invalid))
                } else if (mMessage!!.length == 0) {
                    Utils.showToast(this@ReportProblemActivity, "Please enter the message")
                } else {
                    if (Utils.isNetworkConnected(this@ReportProblemActivity)) {
                        Utils.initializeAndShow(this@ReportProblemActivity)
                        hitAPI()
                    } else {
                        Utils.showToast(this@ReportProblemActivity, resources.getString(R.string.err_no_internet))
                    }
                }
            }
            R.id.txt_attach -> openDialog("report")
        }
    }

    private fun hitAPI() {
        val mUSerId = RequestBody.create(MediaType.parse("text/plain"), SharedPref.getString(Constants.USER_ID, ""))
        val message = RequestBody.create(MediaType.parse("text/plain"), mMessage)
        ImageUploadClass.imageUpload(fileUpload, mUSerId, message, Constants.FOR_REPORT_PROBLEM, this@ReportProblemActivity)
    }

    fun getImageDetails(fileToUpload: MultipartBody.Part?, mUSerId: RequestBody?, name: String?) {
        fileUpload = fileToUpload
        mFileName = name
        tvImageName!!.visibility = View.VISIBLE
        tvImageName!!.text = name
        // mID = mUSerId;
// hitAPI();
    }

    override fun sendResponse(response: Any, TYPE: Int) {
        Utils.hideProgressDialog()
        try {
            val jsonObject = JSONObject(response.toString())
            if (jsonObject.getString("Status").equals("success", ignoreCase = true)) {
                Utils.showToast(this@ReportProblemActivity, jsonObject.getString("Message"))
                val intent = Intent(this@ReportProblemActivity, MainActivity::class.java)
                startActivity(intent)
                finish()
            } else {
                Utils.showToast(this@ReportProblemActivity, jsonObject.getString("Message"))
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }
}